Sources/Foundation/FileHandle+Async.swift

```diff
@@ -0,0 +1,189 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Darwin
+
+final actor IOActor {
+    func read(from fd: Int32, into buffer: UnsafeMutableRawBufferPointer) async throws -> Int {
+        while true {
+            let amount = Darwin.read(fd, buffer.baseAddress, buffer.count)
```

Comment:
This may need to have an alias; for example, from FileHandle.swift:

```
#if canImport(Darwin)
import Darwin
fileprivate let _read = Darwin.read(_:_:_:)
fileprivate let _write = Darwin.write(_:_:_:)
fileprivate let _close = Darwin.close(_:)
#elseif canImport(Glibc)
import Glibc
fileprivate let _read = Glibc.read(_:_:_:)
fileprivate let _write = Glibc.write(_:_:_:)
fileprivate let _close = Glibc.close(_:)
#endif
```

Reply:
It's probably something like this.

```suggestion
#if canImport(Darwin)
            let read = Darwin.read
#elseif canImport(Glibc)
            let read = Glibc.read
#elseif canImport(CRT)
            let read = CRT._read
#endif
            let amount = Darwin.read(fd, buffer.baseAddress, buffer.count)
```

Reply:
cc @compnerd The docs for CRT say `_read`, is that how it is imported?

Reply:
It would be nice to use `_read` to avoid C4996 (i.e. `-Wdeprecated-declaration`).  The import that you have seems correct.  I can try to kick off a test run of this once this is in a good state.